migration/idempotent/001-base.sql

```diff
@@ -2058,6 +2058,37 @@ data newer than `newer_than` in any metric table.
 Note: See _prom_catalog.mark_series_to_be_dropped_as_unused for context.
 ';
 
+CREATE OR REPLACE FUNCTION _prom_catalog.initialize_current_epoch(ran_at TIMESTAMPTZ)
+    RETURNS BIGINT
+    --security definer to have permissions on ids_epoch table
+    SECURITY DEFINER
+    VOLATILE
+    SET search_path = pg_catalog, pg_temp
+AS $func$
+    DECLARE
+        epoch BIGINT;
+        rows_affected INTEGER;
+    BEGIN
+        epoch := EXTRACT (EPOCH FROM ran_at);
+        UPDATE _prom_catalog.ids_epoch e SET current_epoch = epoch WHERE e.current_epoch = 0;
+        GET DIAGNOSTICS rows_affected = ROW_COUNT;
```

Comment:
- Why use `GET DIAGNOSTICS`? Is something wrong with `UPDATE ... RETURNING`? 
- I think always performing the `SELECT current_epoch` is safer.

Reply:
I have opted to always perform `SELECT current_epoch`, as you suggested.